### PR TITLE
Move helpers to solana-cli-config

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
Other crates would like to use url, keypair, and commitment helpers without importing the entirety of `solana-cli` (#24227)

#### Summary of Changes
Add `ConfigInput` to `solana-cli-config` struct to implement this functionality, and use in `solana-cli`

### CODE REVIEW

File: cli-utils/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    solana_cli_config::Config, solana_sdk::commitment_config::CommitmentConfig, std::str::FromStr,
+};
+
+pub enum SettingType {
+    Explicit,
+    Computed,
+    SystemDefault,
+}
+
+pub fn println_name_value_or(name: &str, value: &str, setting_type: SettingType) {
```
Comment (mvines) [inline cli-utils/src/lib.rs:12]: Move this to `cli-output/`?

File: cli-utils/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+    println!(
+        "{} {} {}",
+        style(name).bold(),
+        style(value),
+        style(description).italic(),
+    );
+}
+
+pub struct ConfigInput {
```
Comment (mvines) [inline cli-utils/src/lib.rs:27]: Move this to `cli-config/`?

**Comment by mvines** [comment]
I'm pretty bearish on "util"/"misc"/"helper" crates in general.  I think there's a way to map all that's in `lib.rs` to existing cli-y crates

File: cli-utils/src/lib.rs (hunk 1 above)
Comment (CriesofCarrots) [reply cli-utils/src/lib.rs:12]: Good call.

File: cli-utils/src/lib.rs (hunk 2 above)
Comment (CriesofCarrots) [reply cli-utils/src/lib.rs:27]: I did start down that path, but then it pulls in a dependency on solana_sdk. The `solana-cli-config` is pretty simple and straightforward right now; only does this:
https://github.com/solana-labs/solana/blob/e14933c54db16812803ec51c96c1a31ff0a6bef5/cli-config/src/lib.rs#L1

Thoughts on the extra bulk in `solana-cli-config`?

Comment (mvines) [reply cli-utils/src/lib.rs:12]: Seems ok to me, solana-sdk is almost certainly already in the dependency graph anyway.  `clap-utils/`, another alternative I thought of, doesn't feel quite right.

**Review by mvines — APPROVED** [review]
ty!

**Pull request merged**